File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/ScheduleAdapterItemHeader.java
package com.mentalmachines.droidcon_boston.views.agenda;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.mentalmachines.droidcon_boston.R;
import eu.davidea.flexibleadapter.FlexibleAdapter;
import eu.davidea.flexibleadapter.items.AbstractHeaderItem;
import eu.davidea.viewholders.FlexibleViewHolder;
import java.util.List;

/**
 * Sticky header for schedule view
 */
public class ScheduleAdapterItemHeader extends AbstractHeaderItem<ScheduleAdapterItemHeader.ViewHolder> {

    private String sessionTime;

    public ScheduleAdapterItemHeader(String sessionTime) {
        this.sessionTime = sessionTime;
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof ScheduleAdapterItemHeader) {
            ScheduleAdapterItemHeader inItem = (ScheduleAdapterItemHeader) o;
            return this.sessionTime.equals(inItem.sessionTime);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return sessionTime.hashCode();
    }

    @Override
    public int getLayoutRes() {
        return R.layout.schedule_item_header;
    }

    @Override
    public ScheduleAdapterItemHeader.ViewHolder createViewHolder(FlexibleAdapter adapter,
            LayoutInflater inflater,
            ViewGroup parent) {
        return new ScheduleAdapterItemHeader.ViewHolder(
                inflater.inflate(getLayoutRes(), parent, false), adapter, true);
    }

    @Override
    public void bindViewHolder(FlexibleAdapter adapter,
            ScheduleAdapterItemHeader.ViewHolder holder,
            int position,
            List payloads) {

        holder.header.setText(sessionTime);
    }


    public static class ViewHolder extends FlexibleViewHolder {

        TextView header;

        public ViewHolder(View view, FlexibleAdapter adapter) {
            super(view, adapter);

            findViews(view);
        }

        public ViewHolder(View view, FlexibleAdapter adapter, boolean stickyHeader) {
            super(view, adapter, stickyHeader);

            findViews(view);
        }

        private void findViews(View parent) {
            header = (TextView) parent.findViewById(R.id.header_text);
        }
    }
}

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/ScheduleAdapterItemHeader.kt
package com.mentalmachines.droidcon_boston.views.agenda

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mentalmachines.droidcon_boston.R
import eu.davidea.flexibleadapter.FlexibleAdapter
import eu.davidea.flexibleadapter.items.AbstractHeaderItem
import eu.davidea.flexibleadapter.items.IFlexible
import eu.davidea.viewholders.FlexibleViewHolder

/**
 * Sticky header for schedule view
 */
class ScheduleAdapterItemHeader internal constructor(private val sessionTime: String) :
    AbstractHeaderItem<ScheduleAdapterItemHeader.ViewHolder>() {

    override fun equals(other: Any?): Boolean {
        if (other is ScheduleAdapterItemHeader) {
            val inItem = other as ScheduleAdapterItemHeader?
            return this.sessionTime == inItem!!.sessionTime
        }
        return false
    }

    override fun hashCode(): Int {
        return sessionTime.hashCode()
    }

    override fun getLayoutRes(): Int {
        return R.layout.schedule_item_header
    }

    override fun createViewHolder(
        view: View,
        adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>
    ): ViewHolder {
        return ScheduleAdapterItemHeader.ViewHolder(view, adapter, true)
    }

    override fun bindViewHolder(
        adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>,
        holder: ScheduleAdapterItemHeader.ViewHolder,
        position: Int,
        payloads: MutableList<Any>
    ) {
        holder.header.text = sessionTime
    }

    class ViewHolder : FlexibleViewHolder {

        lateinit var header: TextView

        constructor(view: View, adapter: FlexibleAdapter<*>) : super(view, adapter) {
            findViews(view)
        }

        internal constructor(
            view: View,
            adapter: FlexibleAdapter<*>,
            stickyHeader: Boolean
        ) : super(view, adapter, stickyHeader) {
            findViews(view)
        }

        private fun findViews(parent: View) {
            header = parent.findViewById(R.id.header_text)
        }
    }
}
